### ISSUE TITLE
Add `SymbolicValue` class

### ISSUE THREAD

**Issue by TurtlePU** [zkFold/zkfold-base#308]
Add the following class in new module `ZkFold/Symbolic/Value.hs`:
```haskell
class SymbolicData d => SymbolicValue d where
  type Value d :: Type
  ltv :: (Support d -> Layout d (BaseField (Context d))) -> Value d
  vtl :: Value d -> Support d -> Layout d (BaseField (Context d))
```
The intended meaning of `Value d` is the usual Haskell type which corresponds to Symbolic datatype `d`.
Benefits:
- Better `symbolic*` family of combinators
- Better testing for Symbolic datatypes (e.g. we could use `Arbitrary (Value d)` instead of `toss`es)
- After #307, we can write down `ToJSON` and the like abstractly without specifying the `Interpreter` using:
```haskell
value :: (SymbolicValue d, Context d ~ c, BaseField c ~ WitnessField c) => d -> Value d
value = ltv . fmap witness . pieces
```
- [ ] Add the class definition with common instances
- [ ] Add default implementation via `Generic`
- [ ] Add new `symbolic*` combinators instead of `symbolicF`/`symbolicD` ones
- [ ] Migrate tests to use `SymbolicValue`
- [ ] After #307, migrate explicit `Interpreter` usages to `value`

**Comment by TurtlePU** [comment]
An alternative solution would be to add members of this `SymbolicValue` class back to `SymbolicData`. This way we can also avoid writing doen a single Generic instance of `SymbolicData` and provide only default methods instead, which might look more natural to Haskell developer.

**Comment by TurtlePU** [comment]
Closed in favor of #326
